### DISCUSSION TITLE
[Documentation] Questions

### DISCUSSION THREAD

**Discussion by zardoru** [TeamRizu/OutFox#229] [03 - Feature Requests]
These are questions I'd want answered in documentation. These are assuming that Lua for SM5 exists. If it didn't, add those topics too.

How do I define a new screen, from scratch?
Where do I find an existing screen?
How do I manipulate individual elements in a screen?
What do the different metrics do?
How do I code a noteskin? What can lua do in noteskins? Does it have a particular way of slicing textures? How do they relate to metrics if they exist?
How do I manipulate the playing field?
How do I add things to the playing field through the theme? How do I do it through a song?
What are background events? Foreground events? How do I plug them to lua? Can that be done?
Where is the most up to date documentation that pertains to outfox's lua api, specifically?
How do I respond to events? Button presses? Mouse clicks? Pad inputs?
How can I make stateful things? Commands seem to be a fire-and-forget deal.

**Comment by JoseVarelaP** [comment]
Here goes a checklist of the items mentioned on this list.

### Legacy
x = Done
T = In Planning
W = In the works

- [x] How do I define a new screen, from scratch? (https://outfox.wiki/dev/theming/Theming-2-Screen-Creation/)
- [ ] (T) Where do I find an existing screen?
- [ ] How do I manipulate individual elements in a screen?
- [ ] What do the different metrics do?
- [ ] How do I code a noteskin? What can lua do in noteskins? Does it have a particular way of slicing textures? How do they relate to metrics if they exist?
- [ ] (T) How do I manipulate the playing field?
- [ ] How do I add things to the playing field through the theme? How do I do it through a song?
- [ ] (T) What are background events? Foreground events? How do I plug them to lua? Can that be done?
- [ ] (T) Where is the most up to date documentation that pertains to outfox's lua api, specifically?
*(This one will take a bit given the lua api page is getting a full rework)*

- [x] How do I respond to events? Button presses? Mouse clicks? Pad inputs? (https://outfox.wiki/dev/theming/Theming-Custom-Input/)
- [ ] How can I make stateful things? Commands seem to be a fire-and-forget deal.

**Comment by Sudospective** [comment]
I'm curious about the last question; what plans do you have when it comes to "stateful things"? Judging my the question above it, I want to assume something like a finite state machine, is there some sort of input mechanic you're planning to develop into a theme or chart?
